src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -210,6 +212,126 @@ function digest_boundary_conditions(boundary_conditions::AbstractArray, mesh, so
     throw(ArgumentError("Please use a (named) tuple instead of an (abstract) array to supply multiple boundary conditions (to improve performance)."))
 end
 
+# No checks for these meshes yet available
+function check_periodicity_mesh_boundary_conditions(mesh::Union{P4estMesh,
+                                                                UnstructuredMesh2D,
+                                                                T8CodeMesh,
+                                                                DGMultiMesh},
+                                                    boundary_conditions)
+end
+
+# No actions needed for periodic boundary conditions
+function check_periodicity_mesh_boundary_conditions(mesh::Union{TreeMesh,
+                                                                StructuredMesh},
+                                                    boundary_conditions::BoundaryConditionPeriodic)
+end
+
+function check_periodicity_mesh_boundary_conditions(mesh::Union{TreeMesh{1},
+                                                                StructuredMesh{1}},
+                                                    boundary_conditions::NamedTuple)
+    @unpack x_neg, x_pos = boundary_conditions
+    if isperiodic(mesh, 1) &&
+       (x_neg != BoundaryConditionPeriodic() ||
+        x_pos != BoundaryConditionPeriodic())
+        @error "For periodic mesh non-periodic boundary conditions are supplied.
+         The boundary conditions will be ignored and periodic boundary conditions are used."
+    end
+end
+
+function check_periodicity_mesh_boundary_conditions(mesh::Union{TreeMesh{1},
+                                                                StructuredMesh{1}},
+                                                    boundary_conditions::NTuple{2, Any})
+    if isperiodic(mesh, 1) &&
```

Comment:
Do we need the specializations on `NamedTuple` and `NTuple`? Would it also work without the dispatch when using the version written for `NTuple`?

Reply:
Looks good, lets see if all the tests run through